### ISSUE TITLE
Is there a way to pass initial text and/or immediately return when there is one result?

### ISSUE THREAD

**Issue by NullVoxPopuli** [talmobi/node-fzf#8]
For example, with the bash version of fzf, I can do this: 

```bash
    selected_repo=$(print_repos | fzf \
      -1 \
      --with-nth=1 \
      $QUERY \
      | cut -f2 \
    )
```
the options in particular I'm after are `-1` and `-with-nth=1`

**Comment by talmobi** [comment]
I'm not sure what you're trying to do, but `node-fzf` waits for user input before it does anything. No automatic selection.

**Comment by NullVoxPopuli** [comment]
yeah -- I'm using node-fzf as part of a wrapper cli, and native `fzf` supports passing a starting `$QUERY` to pre-filter the list, so it'd be great to have that here. would improve ergonomics :)

**Comment by talmobi** [comment]
if you're using nfzf JS API you should be able to do it, but the default `nfzf` CLI itself is a very basic implementation.

**Comment by NullVoxPopuli** [comment]
how would you do it? 🤔

**Comment by talmobi** [comment]
you mean

1) pass in initial filter input (as if user already typed it in?)

2) when list only has 1 item after filtering, return it immediately without user having to press Enter key?

**Comment by NullVoxPopuli** [comment]
yes

**Comment by talmobi** [comment]
Can you tell me the use case for 2)? I'm not sure I like it.

**Comment by talmobi** [comment]
@NullVoxPopuli 

1) has been published to v0.11.0 by settings ~`options.prefill`~ `options.query`

[EDIT]: variable renamed to `options.query` and cli args `-q, --query` to match `fzf`

**Comment by ZeroDragon** [comment]
I think I get use case 2
Currently I'm using node-fzf to match if all elements of a playlist are inside my music database.
So, for one side I have a list of songs with their artists. And I have to match if my elements from the playlist are matching.

for each song I do a simple search with data from the playlist vs the data from the database and have a very high percentage of correct matches but there are some that have a slight different name from the playlist vs the database and those return a "mismatch" that I have to search manually (thats where I use node-fzf).

But if instead of doing a "dumb" search using `includes` I could just call node-fzf and automatically return the match only in case that there is only and exactly 1 match, that would make my process even easier for the user.

lets say im looking for `Def Leppard - immigrant song`
and in my database I have `Def Leppard - immigrant song` and `Def Leppard - immigrant song (live)` That would return two options, then a user must define witch one is the correct

but If I have `blink 182 - all the small things` and in my db I have `blink-182 - all the small things` That would mismatch from my dumb search but will return 1 result using fzf. So If I could just start fzf with prefilled text `artist song` and when fzf returns only 1 result, it means that is the correct result and automatically return it, no need for user input.

Then just query the user for those cases where could not resolve into one single result, or when the song is not in the database

Anyway that is just my personal use.

**Comment by talmobi** [comment]
@NullVoxPopuli @ZeroDragon hello and thank you~

--select-1 cli arg (and selectOne option) pushed to master branch (not published to npm yet) -- can you try it and see if it resolves your use case? thank you

**Comment by ZeroDragon** [comment]
https://github.com/user-attachments/assets/35f09691-8b0c-4a73-9b1c-ea4b917d928b

I can confirm it works as expected.
in the screencast, some songs did not autmatically match, but were passed by the automatic fzf with query `artist title` and selectOne = true. Those that only returned 1 (no more, no less) match were automatically returned and processed as if user would select them

**Comment by talmobi** [comment]
@ZeroDragon thanks~ published to npm as latest version `0.13.1
cc @NullVoxPopuli 

let me know if you have issues or suggestions thanks

**Comment by ZeroDragon** [comment]
awesome
